### PR TITLE
[ADD][14.0] edi_webservice_oca_response 

### PR DESCRIPTION
**Pull request by petrus-v**
This module store webservice responses while using webservice in edi framework.

This can be usefull to understand what's wrong while some webservice return usefull informations within http error code.

### CODE REVIEW

File: edi_webservice_oca_response/models/edi_exchange_record.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    ws_response_content = fields.Binary(
+        "Response", attachment=True, copy=False, help="Web service response content"
+    )
+    ws_response_content_filename = fields.Char(
+        compute="_compute_ws_response_content_filename"
+    )
+
+    def _compute_ws_response_content_filename(self):
+        for rec in self:
+            rec.ws_response_content_filename = _("response_") + rec.exchange_filename
```
Comment (simahawk) [inline edi_webservice_oca_response/models/edi_exchange_record.py]: it's a tech name, it should not be translatable IMO

File: edi_webservice_oca_response/components/send.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+from odoo.addons.component.core import Component
+
+
+class EDIWebserviceSend(Component):
+    _inherit = "edi.webservice.send"
+
+    def __init__(self, work_context):
+        super().__init__(work_context)
+        self.ws_settings = getattr(work_context, "webservice", {})
```
Comment (simahawk) [inline edi_webservice_oca_response/components/send.py:15]: this is already defined

**Review by simahawk — CHANGES_REQUESTED** [review]
Hi @petrus-v , thanks for your contrib.

I'd prefer to see this in the `webservice` module.
We can have:
1. a flag on the webservice backend to enable this behavior
2. a `webservice.consumer.mixin`  to reuse where needed 
3. use this mixin in `edi_webservice` to extend edi.exchange.record

Food for thoughts: in the long term I'd like to make `rest_log` independent from `base_rest` and use it to track any kind of request or response.

**Comment by petrus-v** [comment]
> Hi @petrus-v , thanks for your contrib.
> 
> I'd prefer to see this in the `webservice` module. We can have:
> 
>     1. a flag on the webservice backend to enable this behavior
> 
>     2. a `webservice.consumer.mixin`  to reuse where needed
> 
>     3. use this mixin in `edi_webservice` to extend edi.exchange.record

thanks for review and advices, I'll take care of them

**Comment by pedrobaeza** [comment]
Shouldn't it be `edi_webservice_response_oca` or just `edi_webservice_response`?

**Comment by petrus-v** [comment]
@pedrobaeza thanks for your review !

As asked by Simone I'll refactor and move all the code present here in `webservice` and `edi_webservice_oca` so I'll close this PR as soon the refactor is ready and would  like to keep in draft in the mean time (I should find time to do that in given weeks)

**Comment by petrus-v** [comment]
It's time to close this PR in favor of https://github.com/OCA/web-api/pull/22 and https://github.com/OCA/edi/pull/787

**Pull request closed without merging**